src/tests/Interop/ArrayMarshalling/StructArray/MarshalStructArrayTest.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using TestLibrary;
+using Xunit;
+
+public static unsafe class MarshalStructArrayTest
+{
+    [SkipOnMono("Mono doesn't support built-in COM interop, which this test requires")]
+    [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsBuiltInComEnabled))]
+    public static void ArrayElementsInStructFreed()
+    {
+        var (underlyingWeakRef, wrapperWeakRef) = CallNative();
+
+        // Try to GC 10 times until the built-in RCW is collected.
+        for (int i = 0; i < 10 && wrapperWeakRef.IsAlive; i++)
+        {
+            GC.Collect();
+        }
+
+        Assert.False(wrapperWeakRef.IsAlive);
+
+        // Run two more GCs, one to collect the ComWrappers MOW and one to collect the underlying weak reference.
```

Comment:
Tests that depend on object getting collected are often unreliable.

It may be better to directly construct IUnknown vtable with just Release method filled in and verify that the Release method was called.

Reply:
We'll still need to rely on the GC above for the built-in RCW, but I should be able to reduce the complexity here.

Reply:
I might be able to do this without any GC requirements. I'll see what I can do.